test/boost/querier_cache_test.cc

```diff
@@ -365,199 +368,208 @@ class test_querier_cache {
     }
 };
 
-SEASTAR_THREAD_TEST_CASE(lookup_with_wrong_key_misses) {
-    test_querier_cache t;
+future<> do_with_querier_cache_test(std::function<void(test_querier_cache&)> func, std::chrono::seconds entry_ttl = 24h, const noncopyable_function<sstring(size_t)>& external_make_value = test_querier_cache::make_value, ssize_t max_memory = std::numeric_limits<ssize_t>::max()) {
+    return do_with_cql_env_thread([func = std::move(func), &external_make_value, entry_ttl, max_memory] (cql_test_env& cql_env) {
+        test_querier_cache t(cql_env.local_db(), external_make_value, entry_ttl, max_memory);
+        func(t);
```

Comment:
I think all these test changes can be dropped, and you can just pass a method to the querier cache here, which always returns `true`, to keep things simple.